inference-engine/src/inference_engine/src/ie_data.cpp

```diff
@@ -145,6 +188,11 @@ void Data::setPrecision(const Precision& precision) {
 }
 
 const SizeVector& Data::getDims() const {
+    if (isDynamic())
+        IE_THROW() << "Cannot return dims for Data with dynamic shapes!";
+    if (tensorDesc.getDims().empty() && tensorDesc.getLayout() != SCALAR) {
+        tensorDesc.setDims(pShape.to_shape());
```

Comment:
???

Reply:
We need it to sync partialShape and tensorDesc